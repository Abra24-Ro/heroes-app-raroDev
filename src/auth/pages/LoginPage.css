.container-login {
  background-image: url("./topography.svg");
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  min-height: 100vh;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
 
 
}

