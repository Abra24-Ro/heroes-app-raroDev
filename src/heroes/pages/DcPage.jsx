import { HeroList } from "../components/HeroList";

export const DcPage = () => {
  return (
    <>
      <h2 className="text-center ">Dc Comics</h2>
      <hr />
      <HeroList publisher="DC Comics" />
    </>
  );
};
