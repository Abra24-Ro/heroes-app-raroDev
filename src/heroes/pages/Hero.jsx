import { Navigate, useNavigate, useParams } from "react-router-dom";
import { getHeroeById } from "../helpers";
import { useMemo } from "react";

export const Hero = () => {
  const { id } = useParams();
  const navigate = useNavigate();

  const hero = useMemo(() => getHeroeById(id), [id]);

  if (!hero) return <Navigate to="/" />;

  const { superhero, publisher, alter_ego, first_appearance, characters } =
    hero;

  const handleReturn = () => {
    navigate(-1); // Regresa a la página anterior
  };

  return (
    <div className="container mt-5">
      <div className="row align-items-center">
        {/* Imagen */}
        <div className="col-md-4">
          <img
            src={`/assets/heroes/${id}.jpg`}
            alt={superhero}
            className="img-fluid rounded-4 shadow animate__animated animate__fadeInLeft"
          />
        </div>

        {/* Detalles */}
        <div className="col-md-8">
          <h2 className="fw-bold text-primary mb-3">{superhero}</h2>
          <ul className="list-group list-group-flush mb-3">
            <li className="list-group-item">
              <strong>Alter ego:</strong> {alter_ego}
            </li>
            <li className="list-group-item">
              <strong>Primera aparición:</strong> {first_appearance}
            </li>
            <li className="list-group-item">
              <strong>Editorial:</strong> {publisher}
            </li>
            {alter_ego !== characters && (
              <li className="list-group-item">
                <strong>También conocido como:</strong> {characters}
              </li>
            )}
          </ul>

          {/* Botón de regreso */}
          <button
            className="btn btn-outline-primary rounded-pill px-4 mt-3"
            onClick={handleReturn}
          >
            ⬅ Regresar
          </button>
        </div>
      </div>
    </div>
  );
};
