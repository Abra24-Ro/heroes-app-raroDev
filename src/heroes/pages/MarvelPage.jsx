import { HeroList } from "../components/HeroList";

export const MarvelPage = () => {
  return (
    <>
      <h2 className="text-center ">Marvel Comics</h2>
      <hr />
      <HeroList publisher="Marvel Comics" />
    </>
  );
};
