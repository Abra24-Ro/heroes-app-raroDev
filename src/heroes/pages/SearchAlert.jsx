// src/components/SearchAlert.jsx

export const SearchAlert = ({ q = "", heroes = [] }) => {
  if (q === "") {
    return (
      <div className="alert alert-primary animate__animated animate__fadeIn">
        Busca un héroe para ver los resultados
      </div>
    );
  }

  if (heroes.length === 0) {
    return (
      <div className="alert alert-danger animate__animated animate__fadeIn">
        No se encontraron resultados para tu búsqueda
        <strong> {q}</strong>
      </div>
    );
  }

  return null; // No muestra nada si hay resultados
};
