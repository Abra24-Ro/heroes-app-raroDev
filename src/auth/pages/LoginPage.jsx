import { useNavigate } from "react-router-dom";
import "./LoginPage.css";
import { useContext } from "react";
import { AuthContext } from "../context/AuthContext";
import { useForm } from "../../hooks/useForm";

export const LoginPage = () => {
  const { login } = useContext(AuthContext);
  const navigate = useNavigate();
  const { formState, onInputChange} = useForm();

  const handleSubmit = (event) => {
    event.preventDefault();

    if (formState.name.trim().length <= 1) return;

    const lastPath = localStorage.getItem("lastPath") || "/";
    login(formState.name);
    navigate(lastPath, { replace: true });
  };

  return (
    <div className="d-flex justify-content-center align-items-center min-vh-100 container-login">
      <div
        className="card shadow p-4 rounded"
        style={{ maxWidth: "400px", width: "100%" }}
      >
        <div className="card-body">
          <h3 className="card-title text-center mb-4 text-primary">
            Iniciar Sesi√≥n
          </h3>

          <form onSubmit={handleSubmit} >
            <div className="mb-3">
              <label className="form-label">Nombre</label>
              <input
                type="text"
                className="form-control"
                placeholder="Ingrese su nombre"
                name="name"
                value={formState.name}
                onChange={onInputChange}
              />
            </div>

            <button type="submit" className="btn btn-primary w-100">
              Entrar
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};
