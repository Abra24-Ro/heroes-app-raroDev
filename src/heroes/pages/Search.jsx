import { useLocation, useNavigate } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import { HeroItem } from "../components/HeroItem";
import queryString from "query-string";
import { useMemo } from "react";
import { getHeroesByName } from "../helpers";
import { SearchAlert } from "./SearchAlert";


export const Search = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const { q = "" } = queryString.parse(location.search);

  const heroes = useMemo(() => getHeroesByName(q), [q]);

  const { searchText, onInputChange } = useForm({ searchText: q });

  const onSearchSubmit = (event) => {
    event.preventDefault();
    if (searchText.trim().length <= 1) return;
    navigate(`?q=${searchText}`);
  };

  return (
    <div className="container py-4">
      <h2 className="text-center mb-4">ðŸ”Ž Buscar HÃ©roes</h2>
      <hr className="mb-5" />

      <div className="row">
        <div className="col-md-5">
          <div className="card shadow-sm border-0">
            <div className="card-body">
              <h4 className="card-title mb-3">BÃºsqueda</h4>
              <form onSubmit={onSearchSubmit} aria-label="form">
                <input
                  type="text"
                  placeholder="Escribe el nombre de un hÃ©roe..."
                  className="form-control mb-3"
                  name="searchText"
                  autoComplete="off"
                  value={searchText}
                  onChange={onInputChange}
                />
                <button className="btn btn-primary w-100">Buscar</button>
              </form>
            </div>
          </div>
        </div>

        <div className="col-md-7 mt-4 mt-md-0">
          <div className="card shadow-sm border-0">
            <div className="card-body">
              <h4 className="card-title mb-3">Resultados</h4>

                <SearchAlert q={q} heroes={heroes} />

              {heroes.map((hero) => (
                <HeroItem key={hero.id} hero={hero} />
              ))}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
