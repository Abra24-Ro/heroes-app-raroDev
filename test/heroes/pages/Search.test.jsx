import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { describe, expect, test } from "vitest";
import { Search } from "../../../src/heroes/pages/Search";

describe("Pruebas en <Search />", () => {
  test("debe de mostrar el componente correctamente", () => {
    const { container } = render(
      <MemoryRouter>
        <Search />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar a Batman y el input con el valor de query", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <Search />
      </MemoryRouter>
    );

    const input = screen.getByRole("textbox");

    expect(input.value).toBe("batman");

    const img = screen.getByRole("img");
    expect(img.src).toContain("/assets/heroes/dc-batman.jpg");
  });

  test("debe de mostrar un error si no se encuentra el hero", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman123"]}>
        <Search />
      </MemoryRouter>
    );

    expect(
      screen.getByText("No se encontraron resultados para tu bÃºsqueda")
    ).toBeTruthy();
  });

  test("debe de llamar al navigate a la pantalla new", () => {
    render(
      <MemoryRouter initialEntries={["/search"]}>
        <Search />
      </MemoryRouter>
    );

    const input = screen.getByRole("textbox");
    fireEvent.change(input, {
      target: { name: "searchText", value: "batman" },
    });
  });
});
