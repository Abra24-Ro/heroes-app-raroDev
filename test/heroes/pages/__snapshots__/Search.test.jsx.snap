// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en <Search /> > debe de mostrar a Batman y el input con el valor de query 1`] = `
<div>
  <div
    class="container py-4"
  >
    <h2
      class="text-center mb-4"
    >
      ðŸ”Ž Buscar HÃ©roes
    </h2>
    <hr
      class="mb-5"
    />
    <div
      class="row"
    >
      <div
        class="col-md-5"
      >
        <div
          class="card shadow-sm border-0"
        >
          <div
            class="card-body"
          >
            <h4
              class="card-title mb-3"
            >
              BÃºsqueda
            </h4>
            <form>
              <input
                autocomplete="off"
                class="form-control mb-3"
                name="searchText"
                placeholder="Escribe el nombre de un hÃ©roe..."
                type="text"
                value="batman"
              />
              <button
                class="btn btn-primary w-100"
              >
                Buscar
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        class="col-md-7 mt-4 mt-md-0"
      >
        <div
          class="card shadow-sm border-0"
        >
          <div
            class="card-body"
          >
            <h4
              class="card-title mb-3"
            >
              Resultados
            </h4>
            <div
              class="card shadow-sm border-0 rounded-4 h-100 d-flex flex-column animate__animated animate__fadeIn"
              id="dc-batman"
            >
              <img
                alt="Batman"
                class="card-img-top rounded-top-4"
                src="/assets/heroes/dc-batman.jpg"
                style="height: 300px; object-fit: cover;"
              />
              <div
                class="card-body d-flex flex-column "
              >
                <h4
                  class="card-title fw-bold text-primary"
                >
                  Batman
                </h4>
                <h6
                  class="text-muted mb-2"
                >
                  Bruce Wayne
                </h6>
                <p
                  class="mb-1"
                >
                  <strong>
                    ApariciÃ³n:
                  </strong>
                   
                  Detective Comics #27
                </p>
                <p
                  class="mb-0 mt-auto"
                >
                  <strong>
                    Editorial:
                  </strong>
                   
                  DC Comics
                </p>
                <a
                  data-discover="true"
                  href="/hero/dc-batman"
                >
                   Leer mÃ¡s
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <Search /> > debe de mostrar el componente correctamente 1`] = `
<div>
  <div
    class="container py-4"
  >
    <h2
      class="text-center mb-4"
    >
      ðŸ”Ž Buscar HÃ©roes
    </h2>
    <hr
      class="mb-5"
    />
    <div
      class="row"
    >
      <div
        class="col-md-5"
      >
        <div
          class="card shadow-sm border-0"
        >
          <div
            class="card-body"
          >
            <h4
              class="card-title mb-3"
            >
              BÃºsqueda
            </h4>
            <form>
              <input
                autocomplete="off"
                class="form-control mb-3"
                name="searchText"
                placeholder="Escribe el nombre de un hÃ©roe..."
                type="text"
                value=""
              />
              <button
                class="btn btn-primary w-100"
              >
                Buscar
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        class="col-md-7 mt-4 mt-md-0"
      >
        <div
          class="card shadow-sm border-0"
        >
          <div
            class="card-body"
          >
            <h4
              class="card-title mb-3"
            >
              Resultados
            </h4>
            <div
              class="alert alert-primary animate__animated animate__fadeIn"
            >
              Busca un hÃ©roe para ver los resultados
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
